connection_configuration.go

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+
+package gosnowflake
+
+import (
+	"encoding/base64"
+	"errors"
+	"os"
+	path "path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	toml "github.com/BurntSushi/toml"
+)
+
+const (
+	snowflakeConnectionName = "SNOWFLAKE_DEFAULT_CONNECTION_NAME"
+	snowflakeHome           = "SNOWFLAKE_HOME"
+	defaultTokenPath        = "/snowflake/session/token"
+)
+
+// LoadConnectionConfig returns connection configs loaded from the toml file.
+// By default, SNOWFLAKE_HOME(toml file path) is os.snowflakeHome/.snowflake
+// and SNOWFLAKE_DEFAULT_CONNECTION_NAME(DSN) is 'default'
+func loadConnectionConfig() (*Config, error) {
+	cfg := &Config{
+		Params:        make(map[string]*string),
+		Authenticator: AuthTypeSnowflake, // Default to snowflake
+	}
+	dsn := getConnectionDSN(os.Getenv(snowflakeConnectionName))
+	snowflakeConfigDir, err := getTomlFilePath(os.Getenv(snowflakeHome))
+	if err != nil {
+		return nil, err
+	}
+	tomlFilePath := path.Join(snowflakeConfigDir, "connections.toml")
+	err = validateFilePermission(tomlFilePath)
+	if err != nil {
+		return nil, err
+	}
+	tomlInfo := make(map[string]interface{})
+	_, err = toml.DecodeFile(tomlFilePath, &tomlInfo)
+	if err != nil {
+		return nil, err
+	}
+	dsnMap, exist := tomlInfo[dsn]
+	if !exist {
+		return nil, &SnowflakeError{
+			Number:  ErrCodeFailedToFindDSNInToml,
+			Message: errMsgFailedToFindDSNInTomlFile,
+		}
+	}
+	connectionConfig, ok := dsnMap.(map[string]interface{})
+	if !ok {
+		return nil, err
+	}
+	err = parseToml(cfg, connectionConfig)
+	if err != nil {
+		return nil, err
+	}
+	err = fillMissingConfigParameters(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return cfg, err
+}
+
+func parseToml(cfg *Config, connection map[string]interface{}) error {
```

Comment:
connectionConfig?

Reply:
I renamed 'connection to 'connectionMap'